### ISSUE TITLE
Timeout issue when using with Bun

### ISSUE THREAD

**Issue by lnlife** [porsager/postgres#817]
I created a simple http server with `http.createServer`,  and query database when requests come:
`let rows = await sql.unsafe(query, args, { prepare: true })`

After starting the server it works fine at beginning. But if do not send request after a while (maybe 1+ hour), then send request again, got query timeout:
```
22 |       errno: x,
23 |       address: options.path || host
24 |     }, options.path ? {} : { port: port }
25 |   )
26 |   Error.captureStackTrace(error, connection)
             ^
error: write CONNECT_TIMEOUT localhost:5432
 code: "CONNECT_TIMEOUT"

      at connection (/home/test/node_modules/postgres/src/errors.js:26:9)
      at connectTimedOut (/home/test/node_modules/postgres/src/connection.js:433:14)
      at done (/home/test/node_modules/postgres/src/connection.js:1035:12)
```
This issue only occurs on Bun, not on Node.(I've tested both)

**Comment by Nickhoyer** [comment]
I have the same issue. Seems to be fixed in November in #738 but there hasn't been any releases to npm.

@porsager is there any reason there hasn't been any releases to npm?

**Comment by amjed-ali-k** [comment]
@porsager Kindly release new version to npm.  or a beta version.

**Comment by porsager** [comment]
You can install from github by using `porsager/postgres#commithash`

**Comment by porsager** [comment]
> I have the same issue. Seems to be fixed in November in #738 but there hasn't been any releases to npm.
> 
> @porsager is there any reason there hasn't been any releases to npm?

No other reason than limited time on my end 😔

**Comment by porsager** [comment]
v3.4.4 released

**Comment by evelant** [comment]
@porsager a new bug may have been introduced, now when calling `postgres.end({timeout: 30})` it never completes until the timeout on my Bun app

**Comment by porsager** [comment]
please test with node and let me know if there's an issue. If there is only an issue with bun you need to file it in the bun repo.

**Comment by evelant** [comment]
@porsager I'll see if I can reproduce with node, but at least on Bun this is definitely a regression between `3.4.3` and `3.4.4`. I rolled back to `3.4.3` and the connection terminates instantly instead of waiting for the timeout.

**Comment by porsager** [comment]
right, the "workaround!" solving the previous bun error probably shouldn't have been made, and instead solved in bun instead 😅

**Comment by amjed-ali-k** [comment]
@porsager 
seems that "workaround" might be the issue. getting this so frequently now. 


![image](https://github.com/porsager/postgres/assets/86785660/688186be-2a4f-4392-8ebd-4e987c834f33)

**Comment by porsager** [comment]
You should go thumb up that issue on bun ;) https://github.com/oven-sh/bun/issues/7325

**Comment by Jarred-Sumner** [comment]
The fix for this will land in Bun v1.1.7, which we aim to release in a few hours

**Comment by wiesys** [comment]
Hmm... it doesn't work for me even in Bun v1.1.8. Does it work for you @evelant? 🤔

**Comment by evelant** [comment]
I haven't tried it yet, sorry

**Comment by wiesys** [comment]
No worries, in case you have time, you could check so I can be sure that I'm not the only one 🙂

**Comment by dahlia** [comment]
@wiesys I'm using Postgres.js 3.4.4 on Bun 1.1.8 and the issue is still present.

**Comment by amjed-ali-k** [comment]
> The fix for this will land in Bun v1.1.7, which we aim to release in a few hours

Its still there. https://github.com/oven-sh/bun/issues/7325#issuecomment-2123535963

**Comment by amjed-ali-k** [comment]
> The fix for this will land in Bun v1.1.7, which we aim to release in a few hours

@Jarred-Sumner Its still there. https://github.com/oven-sh/bun/issues/7325#issuecomment-2123535963
